### ISSUE TITLE
Documentation around the PREFLIGHT step is needed

### ISSUE THREAD

**Issue by tjtelan** [orbitalci/orbital#155]
Right now I am troubleshooting an error during PREFLIGHT. The errors are not quite intuitive, and I noticed that there's nothing speaking about what this step accomplishes, and that it cannot be skipped. A normal user wouldn't be patient about this failure.

**Comment by shankj3** [comment]
Don't not see anything with `ocelot status`?

**Comment by shankj3** [comment]
Preflight should (hopefully) always mean a descriptive error in `ocelot status`... Should be better documented but hopefully there is troubleshooting there.

**Comment by tjtelan** [comment]
The error from `ocelot status` could be a little more descriptive. It was user error, but all of the logging did not make it obvious enough. The error looked server-side.

This is what the build log output looked like

    $ ocelot logs -hash <hash>
    Warning: partial matching is not yet supported for streaming from storage
    Unable to retrieve from Postgres Database at <postgresdb>.
    Error: sql: no rows in result set

The summary output gives a little more info about internal state

    $ ocelot summary
    +----------+--------+----------------+------------+-----------------+--------+--------+
    | BUILD ID |  REPO  | BUILD DURATION | START TIME |     RESULT      | BRANCH |  HASH  |
    +----------+--------+----------------+------------+-----------------+--------+--------+
       [...]
    |   666    | <repo> |    0 seconds   |     N/A    | FAILED PRESTART | master | <hash> |
       [...]
I imagine pre-build-validation means that at least the ocelot.yml config is valid, since that ended up being the underlying cause of the failures.

And the output from `ocelot status`, which uses PreStart again

    $ ocelot status -hash <hash>
    
    status: Failed PreStart
    hash: <hash>
    account: <account>
    repo: <repo>
    
    [pre-build-validation] took 1 seconds to FAIL

**Comment by tjtelan** [comment]
Failures in PreStart seem to happen out of the box now if the ssh integration is not configured. I'm sure this is new behavior from the refactor, but I'm uncertain as to why. Github support uses ssh keys, so its possible that the vcs setup isn't skipping setting up integrations for Bitbucket, and is setting up for Github.

I have a branch where I had to trace through this issue. I ended up adding a lot of logging and short circuiting in order to get a successful build
https://github.com/tjtelan/orbitalci/commit/3a0a0d59fadaead7abddf5cf1cfffa27051bcd64

Related to #218 

Path to Preflight:
* Werker sees a message for a new build
* The type of werker does a switch between ssh, exec, or docker builder, and then starts that off via `MakeItSo`
* Cleanup functions are deferred
* Consul is written to for Werker bookkeeping.
* The SETUP stage runs. Container starts, and we download a tarball from the L11 S3 bucket
* PREFLIGHT starts

Within Preflight, we load up environment vars from Vault, Download binaries, Run integrations, and then download the codebase -- in this order. There are a bunch of areas where we can fail silently, so extra logging needs to be added. When we fail, we don't clean up our builds. They stay in Running state, so we should have a defer that identifies that condition, and sets status appropriately.
